reVX/plexos/rev_reeds_plexos.py

```diff
@@ -1,1460 +1,1134 @@
+
+        return fmap
+
+    def _make_node_map(self, k=1):
+        """Run ckdtree to map built rev SC points to plexos nodes.
+
+        Parameters
+        ----------
+        k : int
+            Number of neighbors to return.
+
+        Returns
+        -------
+        i : np.ndarray
+            (n, k) array of node indices mapped to the SC builds where n is
+            the number of SC points built and k is the number of neighbors
+            requested. Values are the Plexos node index.
+        """
+
+        plexos_coord_labels = get_coord_labels(self._plexos_nodes)
+        sc_coord_labels = get_coord_labels(self._sc_build)
+        tree = cKDTree(self._plexos_nodes[plexos_coord_labels])
+        d, i = tree.query(self._sc_build[sc_coord_labels], k=k)
+        logger.info('Plexos Node KDTree distance min / mean / max: '
+                    '{} / {} / {}'
+                    .format(np.round(d.min(), decimals=3),
+                            np.round(d.mean(), decimals=3),
+                            np.round(d.max(), decimals=3)))
+
+        if len(i.shape) == 1:
+            i = i.reshape((len(i), 1))
+
+        return i
+
+    def _ammend_plexos_meta(self, i, res_gids, gen_gids, res_built):
+        """Ammend the plexos meta dataframe with data about resource buildouts.
+
+        Parameters
+        ----------
+        i : int
+            Index location to modify (iloc).
+        res_gids : list
+            List of resource GID's that were built for this plexos node.
+        gen_gids : list
+            List of generation GID's that were built for this plexos node.
+        res_built : list
+            List of built capacities at each resource GID for this plexos node.
+        """
+
+        index = self._plexos_meta.index.values[i]
+
+        if self._plexos_meta.loc[index, 'res_gids'] is None:
+            self._plexos_meta.loc[index, 'res_gids'] = str(res_gids)
+            self._plexos_meta.loc[index, 'gen_gids'] = str(gen_gids)
+            self._plexos_meta.loc[index, 'res_built'] = str(res_built)
+
+        else:
+            a = json.loads(self._plexos_meta.loc[index, 'res_gids']) + res_gids
+            b = json.loads(self._plexos_meta.loc[index, 'gen_gids']) + gen_gids
+            c = (json.loads(self._plexos_meta.loc[index, 'res_built'])
+                 + res_built)
+
+            self._plexos_meta.loc[index, 'res_gids'] = str(a)
+            self._plexos_meta.loc[index, 'gen_gids'] = str(b)
+            self._plexos_meta.loc[index, 'res_built'] = str(c)
+
+    def _make_profiles(self):
+        """Make a 2D array of aggregated plexos gen profiles.
+
+        Returns
+        -------
+        output : np.ndarray
+            (t, n) array of Plexos node generation profiles where t is the
+            timeseries length and n is the number of plexos nodes.
+        """
+
+        profiles = self._init_output()
+        progress = 0
+
+        if self.parallel:

```

Comment:
For consistency I think we should use our max_workers kwarg with parallel = max_workers > 1

Reply:
agreed